File: turborepo/packages/api-client/api/email/email-api.ts
import type { ResponseModel } from '@repo/common/model/response-model'
import { http } from '../../core/http'

const BASE_URL = '/email'

export const EmailApi = {
  sendAuthCode: (email: string, type: string) =>
    http.post<ResponseModel<void>>(`${BASE_URL}/send-auth-code`, { email, type }),

  verify: (email: string, code: string, type: string) =>
    http.post<ResponseModel<void>>(`${BASE_URL}/verify`, { email, code, type }),
}

File: mobile-monorepo/packages/api_client/lib/src/api/email_api.dart
import 'package:common/common.dart';

import '../core/api_request_options.dart';

abstract class EmailApi {
  Future<ResponseModel<void>> sendAuthCode(
    SendAuthCodeRequest request, {
    ApiRequestOptions? options,
  });

  Future<ResponseModel<void>> verify(
    VerifyEmailRequest request, {
    ApiRequestOptions? options,
  });
}
